### ISSUE TITLE
Function signature not handled properly when modified by proc-macro

### ISSUE THREAD

**Issue by kinnison** [rust-analyzer/rust-analyzer#7119]
I use the `fehler` crate in a number of projects, and indeed work on a number of other projects which have similar proc-macros which adjust function signatures.

Here is a simple example of a `fehler::throws` use where the type inlay for the return is clearly broken:

![bad](https://user-images.githubusercontent.com/1469421/103436657-12015200-4c16-11eb-8560-754bf5755da9.png)

As you can see, the inlay hint is working on the assumption that the return type of the function `openit()` is `File` whereas in reality it'll be `Result<File, io::Error>` and as a result the inlay is showing `<File as Try>::Ok` as the result of the `?` operator, rather than `File` as it should be.

I can confirm I have `loadOutDirsFromCheck` and `procMacro.enable` turned on, and I have tested this with ff0c976979ad236e725a808fd2dba5e41ff4eace which is HEAD at time of writing this report.

<details> <summary>Reproduction notes</summary>

`Cargo.toml` snippet:

```toml
[dependencies]
fehler = "1"
```

`main.rs`:

```rust
use fehler::throws;
use std::fs::File;

#[throws(std::io::Error)]
fn openit() -> File {
    std::fs::File::open("/oughttofail")?
}

#[throws(Box<dyn std::error::Error>)]
fn main() {
    let _foobar = openit()?;
}
```

If things were correct, the type hint inlay for `_foobar` would be `File` rather than `<File as Try>::Ok`.

For reference, it doesn't look like the generated code is particularly unpleasant to deal with either.  From `cargo expand` the `openit()` function is rewritten as:

```rust
fn openit() -> ::core::result::Result<File, std::io::Error> {
    <_ as ::fehler::__internal::_Succeed>::from_ok(std::fs::File::open("/oughttofail")?)
}
```
</details>

**Comment by lnicola** [comment]
Attribute proc macros are not supported yet, so we use the return type as written down. Then we don't fail to detect that File doesn't actually implement Try and show it as a projection.

**Comment by kinnison** [comment]
@lnicola Thanks for that info.  Is there an issue I can track for progress on attribute proc-macros?

**Comment by lnicola** [comment]
Yes, but I don't have it at hand. It should show up in search.

**Comment by lnicola** [comment]
E.g. https://github.com/rust-analyzer/rust-analyzer/issues/6029

**Comment by flodiebold** [comment]
Let's close this as a duplicate of #6029.
